docs/examples/generatedCanonicalFields.json

```diff
@@ -0,0 +1,822 @@
+    "shortCode": "Org Lobbying Activities",
+    "dataType": "string"
+  },
+  {
+    "label": "Org Start Date",
+    "shortCode": "Org Start Date",
+    "dataType": "string"
+  },
+  {
+    "label": "Org Geographic Areas",
+    "shortCode": "Org Geographic Areas",
+    "dataType": "string"
+  },
+  {
+    "label": "Org Audited Financial Statements",
+    "shortCode": "Org Audited Financial Statements",
+    "dataType": "string"
+  },
+  {
+    "label": "Org Grant Agreement Signatory",
+    "shortCode": "Org Grant Agreement Signatory",
+    "dataType": "string"
+  },
+  {
+    "label": "Org Fiscal Year End Date",
+    "shortCode": "Org Fiscal Year End Date",
+    "dataType": "string"
+  },
+  {
+    "label": "Proposal Name(s)",
+    "shortCode": "Proposal Name(s)",
+    "dataType": "string"
+  },
+  {
+    "label": "Proposal Primary Contact Name",
+    "shortCode": "Proposal Primary Contact Name",
+    "dataType": "string"
+  },
+  {
+    "label": "Proposal Primary Email",
+    "shortCode": "Proposal Primary Email",
+    "dataType": "string"
+  },
+  {
+    "label": "Proposal Primary Phone Number",
+    "shortCode": "Proposal Primary Phone Number",
+    "dataType": "string"
+  },
+  {
+    "label": "Proposal Summary",
+    "shortCode": "Proposal Summary",
+    "dataType": "string"
+  },
+  {
+    "label": "Proposal Budget",
+    "shortCode": "Proposal Budget",
+    "dataType": "string"
+  },
+  {
+    "label": "Proposal Investment Start Date/Duration",
+    "shortCode": "Proposal Investment Start Date/Duration",
+    "dataType": "string"
+  },
+  {
+    "label": "Proposal Investment End Date",
+    "shortCode": "Proposal Investment End Date",
+    "dataType": "string"
+  },
+  {
+    "label": "Proposal Project Total Budget",
+    "shortCode": "Proposal Project Total Budget",
+    "dataType": "number"
+  },
+  {
+    "label": "Proposal Type",
+    "shortCode": "Proposal Type",
+    "dataType": "string"
+  },
+  {
+    "label": "Proposal Fiscal Sponsorship Org Name",
+    "shortCode": "Proposal Fiscal Sponsorship Org Name",
+    "dataType": "string"
+  },
+  {
+    "label": "Proposal Location of Work (free-form string)",
+    "shortCode": "Proposal Location of Work (free-form string)",
+    "dataType": "string"
+  },
+  {
+    "label": "Proposal Involves Lobbying or Electioneering?",
+    "shortCode": "Proposal Involves Lobbying or Electioneering?",
+    "dataType": "string"
+  },
+  {
+    "label": "IRB review needed?",
+    "shortCode": "IRB review needed?",
+    "dataType": "boolean"
+  },
+  {
+    "label": "IRB approval obtained?",
+    "shortCode": "IRB approval obtained?",
+    "dataType": "string"
+  },
+  {
+    "label": "Are sub-grants being given?",
+    "shortCode": "Are sub-grants being given?",
+    "dataType": "string"
+  },
+  {
+    "label": "Sub-grantees",
+    "shortCode": "Sub-grantees",
+    "dataType": "string"
+  },
+  {
+    "label": "Accounting Period End Month",
+    "shortCode": "fiscal_month",
```

Comment:
This is the first Funder-supplied field, so you see the script currently takes the Funder Technical Field Name rather than the (empty) PDC Field Name as the shortCode. For this funder it is more human-friendly, another funder uses universally-unique identifiers.

Reply:
I would suggest we NOT determine the short code based on the funder's technical field names (they'll be mapping those to our short codes either way, and better for us to be consistent in our formatting)

Reply:
I am on board. Maybe the automated pass can limit the word or character count (so we do not end up with "whatStepsDoYouTakeOrProcessesDoYouHaveInPlaceToEnsureThatTheIndividualsOrOrganizationsThatYouMayPayWithMacArthurGrantFundsOtherThanYourOwnStaffAreNotOnTerroristWatchListsArePaidInComplianceWithApplicableAntiMoneyLaunderingStatutesAndTheFundsWillNotBeUsedForCriminalOrIllegalActivity" as a short code) before a manual pass.

Reply:
2338adb has the fix.